apps/ex_wire/lib/ex_wire/sync.ex

```diff
@@ -281,23 +284,26 @@ defmodule ExWire.Sync do
           Blocktree.t(),
           Blocktree.t(),
           Trie.t(),
-          BlockQueue.t(),
-          integer()
-        ) :: {integer(), Trie.t()}
+          BlockQueue.t()
+        ) :: Trie.t()
   defp maybe_request_new_block_or_save(
          block_tree,
          next_block_tree,
          trie,
-         block_queue,
-         last_requested_block
+         block_queue
        ) do
-    # We can make this better, but it's basically "if we change, request another block"
     if block_tree != next_block_tree do
+      # Let's consider saving the trie if it's changed
```

Comment:
How would you feel about removing this comment since the function name seems sufficient?